src/yoroi-wallets/cardano/ShelleyWallet.ts

```diff
@@ -386,11 +388,24 @@ export class ShelleyWallet implements WalletInterface {
     return this.storage.setItem('data', this.toJSON())
   }
 
+  async remove() {
+    await this.clearMemos()
+    await this.clear()
+  }
+
   async clear() {
-    await this.transactionCache?.clear()
+    await this.transactionManager.clear()
     await this.utxoManager.clear()
   }
 
+  saveMemo(txId: string, memo: string): Promise<void> {
+    return this.transactionManager.saveMemo(txId, memo)
+  }
+
+  clearMemos(): Promise<void> {
```

Comment:
what is this used for?

Reply:
To remove the memos when a wallet is removed

Reply:
move this into `remove()`
updates types